storage/localstore/mode_put_test.go

```diff
@@ -318,6 +318,47 @@ func TestModePut_sameChunk(t *testing.T) {
 	}
 }
 
+// TestModePutSync_addToGc validates ModePut* with PutSetCheckFunc stub results
+// in the added chunk to show up in GC index
+func TestModePut_addToGc(t *testing.T) {
+	// PutSetCheckFunc will tell localstore to always Set the chunks that enter
+	opts := &Options{PutSetCheckFunc: func(_ []byte) bool { return true }}
+	for _, m := range []chunk.ModePut{
```

Comment:
This test should contain checks that chunk is not added to the GC if PutSetCheckFunc returns false.

It also should test chunks that are put to the database, but already accessed. It is important that both access and gc indexes are properly updated in this case.